docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: passwordRealm-members
+        identity:
+          username-password:
+             username: aUserNameOfYourChoice
+             password: PutAPasswordHere
+    # map the member protocol authentication to the defined realm
+    member-authentication:
+      realm: passwordRealm-members
+----
+
+[[client-to-member-authn]]
+=== Client-to-member authentication
+
+We can use a **simple** authentication method for clients. It allows
+fine-grained control over the role assignment. As this realm only verifies
+incoming client requests, there is no need to specify the member’s identity
+in the realm configuration.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: simpleRealm-clients
+        authentication:
+          simple:
+            users:
+              - username: root
+                password: 7931verySecret
+                roles:
+                  - admin
+              - username: readonly
+                password: anotherSecret654
+                roles:
+                  - monitor
+    # map the client protocol authentication to the defined realms
+    client-authentication:
+      realm: simpleRealm-clients
+    client-permissions:
+      all:
+        principal: admin
+----
+
+[[client-access-control-basics]]
+== Client access control
+
+When the security is enabled, permission checks are done on client
+requests. You have to grant permissions to named roles to allow clients
+to execute actions on a member.
+
+NOTE: Operations executed directly on members (in embedded mode) or the ones executed
+within member-to-member communication are without permission checks.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    client-permissions:
+      # local clients with "admin" role assigned have full access to all operations
+      all:
+        principal: admin
+        endpoints:
+          - 127.0.0.1
+      map:
+        # every authenticated client can work with a map named "playground"
+        - name: playground
+          actions:
+            - all
+        # clients with role "monitor" can read from a map named "accounts"
+        - name: accounts
+          principal: monitor
+          actions:
+            - read
+        # clients with role "application" can work with the "accounts" 
+        # map if they come from given network segments
+        - name: accounts
+          principal: application
+          endpoints:
+            - 192.168.1.*
+            - 192.168.2.*
+          actions:
+            - create
+            - destroy
+            - put
+            - read
+            - remove
+            - lock
+----
+
+=== Client permissions
+
+Permission has:
+
+* type (e.g. `map permission`)
+* name - usually the data structure name (e.g. `employees`)
+* actions - operations allowed (e.g. `read`, `write`)
+* endpoints - remote client IP addresses to which the permission is
```

Comment:
IIUC, name, actions, and endpoints are children of type. And this isn't clear here in the list format.

Reply:
I'll add an additional paragraph mentioning the topic.